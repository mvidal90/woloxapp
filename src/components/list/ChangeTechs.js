import React, { useState } from 'react';

export const ChangeTechs = ({ setTechs }) => {

    const [inputValue, setInputValue] = useState('hola');

    const handleSubmit = (e) => {
        e.preventDefault();

        setTechs( techs => [ inputValue ]);
    }

    const handleInputChange = ( e ) => {
        setInputValue(e.target.value);
    }

    return (
        <div className = "container">
            <div className = "row">
                <form onSubmit={ handleSubmit }>
                    <input 
                        type = "Text"
                        placeholder = "Busca tu tecnologÃ­a"
                        value = { inputValue }
                        onChange = { handleInputChange }
                    />
                </form>
            </div>
        </div>
    )
}
