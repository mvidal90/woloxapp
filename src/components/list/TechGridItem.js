import React, { useEffect, useState } from 'react';

export const TechGridItem = ({
    tech,
    year,
    author,
    license,
    language,
    type,
    logo,
    stg,
    favorites,
    setFavorites,
    handleFavorites
}) => {

    const [isFavorite, setIsFavorite] = useState(false);

    const add = () =>{
        setIsFavorite( true );
        if (favorites === null) {
            setFavorites([tech]);
        } else {
            setFavorites([...favorites, tech]);
        }
    };

    const del = () =>{
        setIsFavorite( false );
        setFavorites(favorites.filter(e => e !== tech))
    };

    useEffect(() => {

        if (stg !== null) {
            if (stg.includes( tech )) {
            
                setIsFavorite( true );
            }
        }
    }, [stg, tech])
    
    
    return (
        <div  className="card">
            <div className = "logos-container">
                <img src={ logo } alt={ tech } className = "img-size" />
            </div>
            <div className = "data-container">
                <h4>{ tech } ( { year } )</h4>
                <p><strong>Autor: </strong> { author } </p> 
                <p><strong>Licencia: </strong> { license } </p> 
                <p><strong>Lenguaje: </strong> { language } </p> 
                <p><strong>Tipo: </strong> { type } </p> 
            </div>

            <form onSubmit = { handleFavorites }>
                {
                    isFavorite ?
                        <button className = "btn-tch2" onClick = { del }>Quitar de Favoritos</button>
                    :
                        <button className = "btn-tch" onClick = { add } >Agregar a Favoritos</button>
                }
            </form>

        </div>
    )
}
