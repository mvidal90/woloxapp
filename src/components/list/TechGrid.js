import React, { useState } from 'react';
import { useTechs } from '../../hooks/useTechs';
import { TechGridItem } from './TechGridItem';

export const TechGrid = ( {stg} ) => {

    const { data, loading } = useTechs();
    const [favorites, setFavorites] = useState(JSON.parse(localStorage.getItem('favorites')))

    const handleFavorites = (e) => {

        localStorage.setItem('favorites', JSON.stringify( favorites ))

    }

    return (    
    <>
        { loading && <p>Loading</p> }
        <div className = "wrapper">
        
            {
                data.map( tech => (
                    <TechGridItem 
                        stg = { stg }
                        favorites = { favorites }
                        setFavorites = { setFavorites }
                        handleFavorites = { handleFavorites }
                        key={ tech.tech }
                        { ...tech }
                    />
                )) 
            }
        
        </div>
        <div>
            <p>Total de tecnolog√≠as: {data.length}</p>
        </div>
    </>       
)
}
