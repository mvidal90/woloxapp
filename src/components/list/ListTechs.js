import React, { useEffect } from 'react';
import { ChangeTechs } from './ChangeTechs';
import { TechGrid } from './TechGrid';

export const ListTechs = ({ setHome, setMsg }) => {

    let stg = JSON.parse(localStorage.getItem('favorites'));

    useEffect(() => {
        setHome( false );
        if (!localStorage.getItem('favorites')) {
            setMsg(`Favoritos: 0`);
        } else {
            setMsg(`Favoritos: ${ stg.length }`);
        }
    }, [stg, setHome, setMsg]);

    return (
        <section id ="Techs">
            <div className = "container">
                <div className = "row">
                    <h2> WoloxApp </h2>
                    {/* <ChangeTechs setTechs = { setTechs }/> */}
                    <ol>
                        <TechGrid  stg = { stg }/>
                    </ol>
                </div>
            </div>
        </section>
    );
}
