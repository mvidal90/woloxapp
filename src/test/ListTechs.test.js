import React from 'react';
import { shallow } from 'enzyme';
import { ListTechs } from '../components/list/ListTechs';

describe('Pruebas con <ListTechs />', () => {

    test('Debe mostrarse correctamente', () => {
          
        const wrapper = shallow( <ListTechs /> );

        expect( wrapper ).toMatchSnapshot(); 
        expect( wrapper.find('TechGrid').length).toBe(1);
        expect( wrapper.find('h2').length).toBe(1);
        

    })

})
