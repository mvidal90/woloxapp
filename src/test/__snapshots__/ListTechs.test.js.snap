// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con <ListTechs /> Debe mostrarse correctamente 1`] = `
<section
  id="Techs"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <h2>
         WoloxApp 
      </h2>
      <ol>
        <TechGrid
          stg={null}
        />
      </ol>
    </div>
  </div>
</section>
`;
